python/cudf/dataframe/dataframe.py

```diff
@@ -408,6 +409,127 @@ def __repr__(self):
             len(self),
         )
 
+    # binary, rbinary, unary, orderedcompare, unorderedcompare
+    def _call_op(self, other, internal_fn, fn):
+        result = DataFrame()
+        result.set_index(self.index)
+        if isinstance(other, Sequence):
+            for k, col in enumerate(self._cols):
+                result[col] = getattr(self._cols[col], internal_fn)(
+                        other[k],
+                        fn,
+                )
+        elif isinstance(other, DataFrame):
+            for col in other._cols:
+                if col in self._cols:
+                    result[col] = getattr(self._cols[col], internal_fn)(
+                            other._cols[col],
+                            fn,
+                    )
+                else:
+                    result[col] = Series(cudautils.full(self.shape[0],
+                                         np.dtype('float64').type(np.nan),
+                                         'float64'), nan_as_null=False)
+            for col in self._cols:
+                if col not in other._cols:
+                    result[col] = Series(cudautils.full(self.shape[0],
+                                         np.dtype('float64').type(np.nan),
+                                         'float64'), nan_as_null=False)
+        elif isinstance(other, Series):
```

Comment:
I imagine Index is the same as here, but we should have an `isinstance(other, Index)`block as well.

Reply:
We can do this in a followup PR when we enable ops on Indexes in general though.